uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PortletPrefsRESTController.java

```diff
@@ -0,0 +1,679 @@
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .body(mapper.writeValueAsString(prefs.getMap()));
+        } catch (Exception e) {
+
+            logger.error(e.getMessage(), e);
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Get uPortal definition preferecnces for portlet/soffit/webcomponent. This method provides a
+     * REST interface for uPortal for obtaining the definition preferences for the
+     * portlet/soffit/webcomponent
+     *
+     * <p>The path for this method is /prefs/getdefinitiononlyprefs/fname. The fname is a string
+     * representing the fname of the portlet/soffit/webcomponent to return the definition
+     * preferences. These are the default preferences for the given portlet/soffit/webcomponent.
+     * Will include extra preferences for soffits -including the url- so please set any of these to
+     * read-only when you define them. You must be logged in as a portal administrator or have
+     * configuration rights for the given portlet/soffit/webcomponent in uPortal to access these.
+     * Will return a 403 or 404 otherwise
+     */
+    // must be authenticated and a portal administrator
+    @RequestMapping(value = "/prefs/getdefinitiononlyprefs/{fname}", method = RequestMethod.GET)
+    public ResponseEntity getDefinitionOnlyPrefs(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            @ApiParam(value = "The portlet fname to find preferences for")
+                    @PathVariable(value = "fname")
+                    String fname) {
+        HttpSession session = request.getSession(false);
+        if (session == null) {
+            return new ResponseEntity<>("", HttpStatus.NOT_FOUND);
+        }
+        final IPerson person = personManager.getPerson(request);
+        if (person.isGuest()) {
+            logger.warn("Guest user tried to access getDefinitionPrefs. Possible hacking attempt.");
+            return new ResponseEntity<>(
+                    "ERROR: guest cannot use this action.", HttpStatus.UNAUTHORIZED);
+        } else if (!person.getSecurityContext().isAuthenticated()) {
+            logger.warn(
+                    "Person: {"
+                            + person.getUserName()
+                            + "} tried to access getDefinitionPrefs while not authenticated. Possible hacking attempt.");
+            return new ResponseEntity<>(
+                    "ERROR: must be logged in to use this action.", HttpStatus.UNAUTHORIZED);
+        }
+
+        try {
+            IPortletDefinition portletDefinition = portletDao.getPortletDefinitionByFname(fname);
+            if (portletDefinition == null) {
+                return new ResponseEntity<>("ERROR: Portlet not found", HttpStatus.NOT_FOUND);
+            }
+
+            if (!canConfigure(person, portletDefinition.getPortletDefinitionId())) {
+                logger.warn(
```

Comment:
With SLF4j loggers you can apply a replace form automatically without appending the properties, like 
```java
logger.warn("My text with a prop {} and an other prop {}", prop1, prop2). And you won't have a NPE as it's managed automatically.